src/styles/calendar.scss

```diff
@@ -0,0 +1,54 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $boxGap;
+  padding: $calendarPadding;
+  right: 0;
+  box-sizing: border-box;
+  border: $calendarBorder solid black;
+  width: $dayBoxW * $boxRow + $calendarPadding
+    * 2 + $dayBoxBorder *3 + $boxGap * $boxRow
+  ;
+
+  @for $lastDay from 29 through 31 {
+    &--month-length-#{$lastDay - 1} {
+      & :nth-child(n + #{$lastDay}) {
+        display: none;
+      }
+    }
+  }
+
+  @each $number, $day in $days {
+    &--start-day-#{$day} :nth-child(#{$number}) {
+      margin-left: #{$number}px;
```

Comment:
Consider counting the margin by multiplying the number and the width of the cell + gap